lbrynet/lbrynet_daemon/auth/server.py

```diff
@@ -185,13 +217,23 @@ def expire_session():
         else:
             d = defer.maybeDeferred(function, *args)
 
-        # cancel the response if the connection is broken
-        notify_finish.addErrback(self._response_failed, d)
+        # finished_deferred will fire when the request is finished
+        # this could be because the request is really done, or because the connection dropped
+        # if the connection dropped, cancel the deferred stack
+        # otherwise it'll try writing to a closed request and twisted doesn't like that
+
+        # TODO: don't trap RuntimeError, which is presently done to handle deferredLists that
+        # won't peacefully cancel, namely get_lbry_files
+
+        finished_deferred.addBoth(self._handle_dropped_request, d, function_name)
```

Comment:
Given that `finished_deferred` is "A Deferred which will be triggered when the request is finished -- with a None value if the request finishes successfully or with an error if the request is interrupted by an error (for example, the client closing the connection prematurely)"

It seems worthwhile to treat the two cases differently.  Is there a situation in which finished_deferred will fire  a successful callback without `d` having been called?